content/automate/ManageIQ/Cloud/VM/Provisioning/Placement.class/__methods__/best_fit_google.rb

```diff
@@ -1,27 +1,52 @@
-###################################################################
-#
-# Description: Select the cloud network and availability zone for Google
-#
-###################################################################
-
-prov  = $evm.root["miq_provision"]
-image = prov.vm_template
-raise "Image not specified" if image.nil?
-
-if prov.get_option(:availability_zone).nil?
-  availability_zone = prov.eligible_availability_zones.first
-
-  if availability_zone
-    prov.set_availability_zone(availability_zone)
-    $evm.log("info", "Image=[#{image.name}] Availability Zone=[#{availability_zone.name}]")
-  end
-end
+##########################################################################
+#                                                                        #
+# Description: Select the cloud network and availability zone for Google #
+#                                                                        #
+##########################################################################
+module ManageIQ
+  module Automate
+    module Cloud
+      module VM
+        module Provisioning
+          module Placement
+            class BestFitGoogle
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                @prov = @handle.root["miq_provision"]
```

Comment:
@pkomanek 
I think we should by using a prov method which can check for missing field raise an error similar to what you have in the other file in this PR